test/import/FSHImporter.SDRules.test.ts

```diff
@@ -1319,6 +1319,58 @@ describe('FSHImporter', () => {
         );
       });
 
+      it('should parse an only rule with a CodeableReference to one type', () => {
```

Comment:
If you can confirm that these are exactly the same, I'm fine with removing them. We may have overlapping tests due to previous refactoring... Or in a few cases, I think we have a token test cases in broad test files and then many detailed ones in more specific test files. Maybe that is what's going on here? Either way, duplicate tests aren't really helpful so I'm fine w/ them going away.

That said, if there are lots of duplicate tests in this file in general, then it may be worth making a separate task to clean the whole thing up.

Reply:
I double checked, and as far as I can tell, they are exactly the same. Neither set of tests sets up anything different in `describe` and/or `beforeAll`/`beforeEach` blocks.

There do seem to be large amounts of duplicates between these two files. The `FSHImporter.SDRules.tests.ts` file does mention at the top that it is designed to cover the overlap between the four different StructureDefinition entities (profiles, extensions, resources, and logicals). I looked at the OnlyRule section, where the tests I added were, as well as a few other sections. There is a large amount of overlap between the tests, but unfortunately there are a few tests that were newly added in both files. For example, the CaretValueRule section has 4 tests shared between the two files, 1 new test added in the FSHImporter.Profile file, and 1 new test added in the FSHImporter.SDRules file.

So that is... unfortunate. It seems like the files might have been refactored when Logicals and Resources were added, so I wonder if something got messed up in the rebasing/merging back then. Who knows.

All that said, I think what makes sense is to add the three new tests I included here into the `FSHImporter.SDRules.test.ts` file and remove them from the `FSHImporter.Profile.test.ts` file since they apply to any StructureDefinition and that will reduce clutter for now. And then it probably is a good idea to clean up these files so there are fewer shared tests between them.

Reply:
I checked the InsertRule block and there's a whole bunch of tests in the `FSHImporter.Profile.test.ts` that aren't in the `FSHImporter.SDRules.test.ts` (~12 tests). So I actually don't know for sure which file we should be trying to maintain, but I think the `FSHImporter.SDRules.test.ts` makes more sense as a centralized place to hold the overlapping tests. So I'll still plan to move my tests to be in `FSHImporter.SDRules.test.ts`, but I can be convinced otherwise if either of you disagree on the file to hold them.

Reply:
Removed the duplicate tests in [7d6c098](https://github.com/FHIR/sushi/pull/1292/commits/7d6c098e0d9d0c1a5a45c4093d60edcf674df3b5), but I can swap the location once you have a chance to process the novels I wrote above.

Reply:
I don't know. On one hand, I understand the approach of having tests for logic that applies to any SD being in `FSHImporter.SDRules.test.ts`. On the other hand, when you write the test, you _do_ have to choose a specific kind of SD, right? But when we're talking about importing rules, I guess it shouldn't matter if it is a profile, logical, etc...  So... ???

I think one of the reasons we split into `FSHImporter.Profile.test.ts` and others was to keep `FSHImporter.SDRules.test.ts` from becoming massive. But at the same time, if we put a rule import test _only_ in `FSHImporter.Profile.test.ts`, then it's not easy to know where to look for it and people might thing that type of rule is not tested at all.

All this to say, I think you can justify either way. But I guess maybe if I had to lean, I'd lean toward `FSHImporter.SDRules.test.ts`? Maybe we should refactor into a separate file for every rule type instead. That actually might make more sense, but is way outside the scope of this PR.

Reply:
> On the other hand, when you write the test, you do have to choose a specific kind of SD, right?

Yes, you do. It seems like the majority of the tests in `FSHImporter.SDRules.test.ts` are testing `Profile`, with a handful for `Resource` when testing Resource/Logical Model specific rules.

I was looking at the file history trying to figure out which file came first, and it seemed like SDRules came second, which is why I thought maybe it was trying to ease the number of tests in the Profile file, but that was really just an inference.

> But when we're talking about importing rules, I guess it shouldn't matter if it is a profile, logical, etc... So... ???

Also, when looking at the Importer code, we do specifically have a `visitSdRule` function that is called when importing all four types of StructureDefintions. So maybe to align with the code, the tests in `FSHImporter.SDRules.test.ts` should align with the functionality in `visitSdRule` and any extra functionality, like the rules specific to Logical Models and Resources, are included in separate files for those entities. It feels like maybe that was what we tried to do and things got messy. I'm not sure. And I guess I don't need to be sure for right now.